tests/log_util_test.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import inspect
+import unittest
+
+from perfkitbenchmarker import log_util
+from perfkitbenchmarker import vm_util
+
+
+class LogUtilTestCase(unittest.TestCase):
+  """Tests exercising the utilities in log_util.
+  """
+
+  def testThreadLogContextLabelExtension(self):
+    """Verify ThreadLogContext.LabelExtension behavior.
+    """
+    context = log_util.ThreadLogContext()
+    self.assertEqual(context._label_list, [])
+    self.assertEqual(context.label, '')
+    with context.LabelExtension('LABEL-A'):
+      self.assertEqual(context._label_list, ['LABEL-A'])
+      self.assertEqual(context.label, 'LABEL-A ')
+      with context.LabelExtension('LABEL-B'):
+        self.assertEqual(context._label_list, ['LABEL-A', 'LABEL-B'])
+        self.assertEqual(context.label, 'LABEL-A LABEL-B ')
+      self.assertEqual(context._label_list, ['LABEL-A'])
+      self.assertEqual(context.label, 'LABEL-A ')
+    self.assertEqual(context._label_list, [])
+    self.assertEqual(context.label, '')
+
+  def testThreadLogContextCopyConstruct(self):
+    """Verify ThreadLogContext init with a reference ThreadLogContext behavior.
+
+    The label state of the first ThreadLogContext should be copied.
+    """
+    original = log_util.ThreadLogContext()
+    self.assertEqual(original._label_list, [])
+    self.assertEqual(original.label, '')
+    with original.LabelExtension('LABEL-A'):
+      self.assertEqual(original._label_list, ['LABEL-A'])
```

Comment:
Nit: `_label_list` is an implementation detail; testing the value of `label` should be sufficient.